### PR TITLE
User activation and enforcing user to change password

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

**Comment by JarekW** [comment]
Why `CHANGE_PASSWORD` event is triggered? even before `USER_CREATED` event

**Comment by chives** [comment]
@JarekW because user password is being encoded in the listener of this event

**Comment by JarekW** [comment]
but maybe `UserEncodePasswordListener` will listen also to `USER_CREATED` event (with lover priority?). Now setter for enforsing password change is after change event to prevent listener.
(I dont know this will be a good solution)

File: Entity/FOSUser.php [hunk 1]
```diff
@@ -18,6 +23,29 @@
...
+    {
+        parent::__construct();
+
+        $this->enforcePasswordChange = false;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasEnforcedPasswordChange()
```
Comment (szymach) [inline Entity/FOSUser.php]: this method name doesn't sit well with me - should be isEnforcedPasswordChange I think

File: Resources/doc/events.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Events
```
Comment (szymach) [inline Resources/doc/events.md]: You mixed the brackets in anchors in the whole file - [] with ()

**Comment by JarekW** [comment]
:+1:

**Comment by szymach** [comment]
:+1:

**Comment by rn0** [comment]
:+1:

**Pull request merged**